src/Toolkit/Toolkit.Maui/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -176,4 +178,28 @@ private async void Refresh()
     /// </summary>
     [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
     public event System.EventHandler? SourceUpdated;
+
+    private void SymbolDisplay_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(Window) && Window != null)
+        {
+            Refresh();
+            Window.DisplayDensityChanged += Window_DisplayDensityChanged;
+        }
+    }
+
+    private void SymbolDisplay_Unloaded(object? sender, EventArgs e)
+    {
+        if (Window != null)
+        {
+            Window.DisplayDensityChanged -= Window_DisplayDensityChanged;
+        }
```

Comment:
Better to rely on weak event handler, and then here releasing the weak handler, rather than relying on Window still being around